src/common/utils/requests.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import * as http from 'http';
+import * as https from 'https';
+
+type Method = 'GET' | 'POST' | 'DELETE' | 'PUT' | 'OPTIONS' | 'HEAD';
+
+export function ping(x: URL, method: Method = 'GET'): Promise<http.ServerResponse> {
+    const f = x.protocol === 'http:' ? http.request : https.request;
+    return new Promise((yes, no) => {
```

Comment:
Instead of returning a Promise, we could mark this function as `async`. It's a bit cleaner.

Reply:
Is there any point in that? Since node only provides event handlers for the request and the response objects, we'd have to do manual Promise resolution anyway, and end up generating one more promise than necessary from V8's point of view.

Reply:
I don't expect this function to be called to the extent that the performance gains matters, so if you think it looks cleaner, the few bytes of extra memory are worth the tradeoff.

Reply:
I opted for adding async.